.github/workflows/e2e-testing.yml

```diff
@@ -0,0 +1,48 @@
+name: e2e Testing
+run-name: e2e tests run by ${{ github.actor }}
+on:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - "README.md"
+      - ".gitignore"
+      - "doc/**"
+  pull_request:
+  workflow_dispatch:
+jobs:
+  e2e-testing:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: "^1.20"
+      - name: Install kind
+        run: go install sigs.k8s.io/kind@v0.20.0
+      - name: Create kind cluster with local registry
+        run: ./e2e/deploy/kind-with-registry.sh
+      - name: Install Rust toolchain
+        run: |
+          curl --proto '=https' --tlsv1.2 --retry 10 --retry-connrefused -fsSL "https://sh.rustup.rs" | sh -s -- --default-toolchain none -y
+      - name: Build zookeeper controller image
+        run: |
+          cp docker/zookeeper-controller/Dockerfile .
+          docker build -t localhost:5001/zookeeper-controller:1.0 .
+      - name: Push zookeeper controller image to local registry
+        run: docker push localhost:5001/zookeeper-controller:1.0
+      - name: Build rabbitmq controller image
+        run: |
+          cp docker/rabbitmq-controller/Dockerfile .
+          docker build -t localhost:5001/rabbitmq-controller:1.0 .
+      - name: Push rabbitmq controller image to local registry
+        run: docker push localhost:5001/rabbitmq-controller:1.0
+      - name: Deploy zookeeper controller
+        run: ./e2e/deploy.sh zookeeper
+      - name: Deploy rabbitmq controller
+        run: ./e2e/deploy.sh rabbitmq
+      - name: Run zookeeper e2e tests
+        run: cd e2e && cargo run -- zookeeper
+      - name: Run rabbitmq e2e tests
+        run: cd e2e && cargo run -- rabbitmq
```

Comment:
I would suggest we test the controllers separately: create kind cluster -> test zk -> teardown kind cluster -> create kind cluster -> test rabbitmq -> teardown cluster